pkg/cmd/roachprod-microbench/compare.go

```diff
@@ -318,6 +355,171 @@ func (c *compare) compareUsingThreshold(comparisonResultsMap model.ComparisonRes
+		if err != nil {
+			if !required && oserror.IsNotExist(err) {
+				return nil
+			}
+			return errors.Wrapf(err, "failed to read file %s", filePath)
+		}
+		benchBuf.Write(data)
+		benchBuf.WriteString("\n")
+		return nil
+	}
+
+	for k, v := range c.influxConfig.metadata {
+		benchBuf.WriteString(fmt.Sprintf("%s: %s\n", k, v))
+	}
+
+	logPaths := map[string]string{
+		"experiment": c.experimentDir,
+		"baseline":   c.baselineDir,
+	}
+	for logType, dir := range logPaths {
+		benchBuf.WriteString(fmt.Sprintf("cockroach: %s\n", logType))
+		logPath := filepath.Join(dir, "metadata.log")
+		if err = readFileFn(logPath, true); err != nil {
+			return nil, err
+		}
+		for _, pkg := range c.packages {
+			benchBuf.WriteString(fmt.Sprintf("pkg: %s\n", pkg))
+			logPath = filepath.Join(dir, getReportLogName(reportLogName, pkg))
+			if err = readFileFn(logPath, false); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	benchReader := benchfmt.NewReader(bytes.NewReader(benchBuf.Bytes()), "buffer")
+	recordsMap := make(map[string][]*benchfmt.Result)
+	seen := make(map[string]map[string]struct{})
+	for benchReader.Scan() {
+		switch rec := benchReader.Result().(type) {
+		case *benchfmt.SyntaxError:
+			// In case the benchmark log is corrupted or contains a syntax error, we
+			// want to return an error to the caller.
+			return nil, fmt.Errorf("syntax error: %v", rec)
+		case *benchfmt.Result:
+			var cmp, pkg string
+			for _, config := range rec.Config {
+				if config.Key == "pkg" {
+					pkg = string(config.Value)
+				}
+				if config.Key == opts.Compare {
+					cmp = string(config.Value)
+				}
+			}
+			key := pkg + packageSeparator + string(rec.Name)
+			// Update the name to include the package name. This is a workaround for
+			// `benchseries`, that currently does not support package names.
+			rec.Name = benchfmt.Name(key)
+			recordsMap[key] = append(recordsMap[key], rec.Clone())
+			// Determine if we've seen this package/benchmark combination for both
+			// the baseline and experiment run.
+			if _, ok := seen[key]; !ok {
+				seen[key] = make(map[string]struct{})
+			}
+			seen[key][cmp] = struct{}{}
+		}
+	}
+
+	// Add only the benchmarks that have been seen in both the baseline and
+	// experiment run.
+	for key, records := range recordsMap {
+		if len(seen[key]) != 2 {
+			continue
+		}
+		for _, rec := range records {
+			builder.Add(rec)
+		}
+	}
+
+	comparisons, err := builder.AllComparisonSeries(nil, benchseries.DUPE_REPLACE)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to creation comparison series")
+	}
+	return comparisons, nil
+}
+
+func (c *compare) pushToInfluxDB() error {
+	client := influxdb2.NewClient(c.influxConfig.host, c.influxConfig.token)
+	defer client.Close()
+	writeAPI := client.WriteAPI("cockroach", "microbench")
+	errorChan := writeAPI.Errors()
+
+	comparisons, err := c.createBenchSeries()
+	if err != nil {
+		return err
+	}
+
+	for _, cs := range comparisons {
+		cs.AddSummaries(0.95, 1000)
```

Comment:
Educational question: I *think* I understand the idea behind bootstrap samples, but I'm having trouble understanding what the 1000 samples are of. i.e. if a benchmark metric is sec/op, are we collecting a timeseries of that throughout the benchmark and selecting 1000 random points to construct the summary of?

Or am I completely missing what this is doing here.

Reply:
I think it's more of a `resolution` for a calculation used in `benchseries.go`. For this calculation we only have 10 iterations (or values) for each revision; so no timeseries, only a single point in time. The goal of the calculation is to get a type of confidence variance for the comparison of the two runs per metric. I didn't delve to deep into the logic / algorithm, but followed the typical API usage of `benchseries`.

Reply:
Ah I see, since it's with replacement we can have 1000 samples even with 10 values?